pandas/io/pytables.py

```diff
@@ -5209,7 +5209,11 @@ def _unconvert_string_array(
         dtype = f"U{itemsize}"
 
         if isinstance(data[0], bytes):
-            data = Series(data, copy=False).str.decode(encoding, errors=errors)._values
+            ser = Series(data, copy=False).str.decode(encoding, errors=errors)
+            if get_option("future.infer_string"):
+                data = ser.to_numpy()
+            else:
+                data = ser._values
```

Comment:
You can probably simplify this and always to `.to_numpy()`? (or `np.asarray(..)`) 
In the case of object dtype in the `else` branch, that will return the same (and be as cheap) as `_values` I think

Reply:
Confirmed - thanks.